src/lib/Tisch/Internal/Singletons.hs

```diff
@@ -25,123 +25,129 @@ module Tisch.Internal.Singletons
-data FirstSym1 (l_aYuv :: TyFun x1627622409 x'1627622410 -> Type)
-               (l_aYuu :: TyFun (x1627622409, y1627622411) (x'1627622410, y1627622411))
-  = forall arg_aYuw. KindOf (Apply (FirstSym1 l_aYuv) arg_aYuw) ~ KindOf (FirstSym2 l_aYuv arg_aYuw) =>
+
+type instance Apply (SecondSym1 j) i = SecondSym2 j i
+
+data SecondSym0 (g ::
+   TyFun (TyFun y6 y7 -> Type) (TyFun (x8, y6) (x8, y7) -> Type)) =
+  forall h. KindOf (Apply SecondSym0 h) ~ KindOf (SecondSym1 h) =>
+            SecondSym0KindInference
+
+type instance Apply SecondSym0 g = SecondSym1 g
+
+type FirstSym2 (q :: TyFun x9 x10 -> Type)
+               (r :: (x9, y11)) = First q r
+
+data FirstSym1 (v :: TyFun x9 x10 -> Type)
+               (l :: TyFun (x9, y11) (x10, y11))
+  = forall w. KindOf (Apply (FirstSym1 v) w) ~ KindOf (FirstSym2 v w) =>
     FirstSym1KindInference
-type instance Apply (FirstSym1 l_aYuv) l_aYuu = FirstSym2 l_aYuv l_aYuu
-instance SuppressUnusedWarnings FirstSym0 where
-  suppressUnusedWarnings _ = snd ((,) FirstSym0KindInference ())
-data FirstSym0 (l_aYus :: TyFun (TyFun x1627622409 x'1627622410 -> Type) (TyFun (x1627622409, y1627622411) (x'1627622410, y1627622411) -> Type))
-  = forall arg_aYut. KindOf (Apply FirstSym0 arg_aYut) ~ KindOf (FirstSym1 arg_aYut) =>
+
+type instance Apply (FirstSym1 v) l = FirstSym2 v l
+
+data FirstSym0 (s :: TyFun (TyFun x9 x10 -> Type) (TyFun (x9, y11) (x10, y11) -> Type))
+  = forall a. KindOf (Apply FirstSym0 a) ~ KindOf (FirstSym1 a) =>
     FirstSym0KindInference
-type instance Apply FirstSym0 l_aYus = FirstSym1 l_aYus
-type family (:&&&) (a_aYtK :: TyFun x_aYtd y_aYte -> Type)
-                   (a_aYtL :: TyFun x_aYtd z_aYtf -> Type)
-                   (a_aYtM :: x_aYtd) :: (y_aYte, z_aYtf) where
-  (:&&&) f_aYtP g_aYtQ a_1627622439_aYtR = Apply (Apply (Apply (Apply Lambda_1627622444Sym0 f_aYtP) g_aYtQ) a_1627622439_aYtR) a_1627622439_aYtR
-type family Second (a_aYul :: TyFun y_aYtg y'_aYth -> Type)
-                   (a_aYum :: (x_aYti, y_aYtg)) :: (x_aYti, y'_aYth) where
-  Second f_aYun '(x_aYuo, y_aYup) = Apply (Apply Tuple2Sym0 x_aYuo) (Apply f_aYun y_aYup)
-type family First (a_aYux :: TyFun x_aYtj x'_aYtk -> Type)
-                  (a_aYuy :: (x_aYtj, y_aYtl)) :: (x'_aYtk, y_aYtl) where
-  First f_aYuz '(x_aYuA, y_aYuB) = Apply (Apply Tuple2Sym0 (Apply f_aYuz x_aYuA)) y_aYuB
+
+type instance Apply FirstSym0 s = FirstSym1 s
+
+type family (:&&&) (k :: TyFun d e -> Type)
+                   (l :: TyFun d f -> Type)
+                   (m :: d) :: (e, f) where
+  (:&&&) p q r = Apply (Apply (Apply (Apply LambdaSym0 p) q) r) r
+
+type family Second (l :: TyFun g h -> Type)
+                   (m :: (i, g)) :: (i, h) where
+  Second n '(o, p) = Apply (Apply Tuple2Sym0 o) (Apply n p)
+
+type family First (x :: TyFun j k -> Type)
+                  (y :: (j, l)) :: (k, l) where
+  First z '(a, b) = Apply (Apply Tuple2Sym0 (Apply z a)) b
+
 infixr 3 :&&&
 
+instance SuppressUnusedWarnings (:&&&$) where
```

Comment:
I don't think we need these `SuppressUnusedWarnings` things. Can we remove them or do we get some warning? Also, I think you can delete all of the constructors named `*KindInference`, I don't think we need those.